src/gui/MessageWidget.cpp

```diff
@@ -50,8 +55,14 @@ void MessageWidget::showMessage(const QString& text, KMessageWidget::MessageType
 {
     setMessageType(type);
     setText(text);
-    emit showAnimationStarted();
-    animatedShow();
+    if (m_animate) {
+        emit showAnimationStarted();
+        animatedShow();
+    } else {
+        show();
+        emit showAnimationFinished();
```

Comment:
What happens if something binds to showAnimationStarted()? Shouldn't this be emitted in any case?